core/templates/dev/head/pages/exploration-player-page/services/exploration-engine.service.ts

```diff
@@ -81,7 +84,20 @@ angular.module('oppia').factory('ExplorationEngineService', [
     // Param changes to be used ONLY in editor preview mode.
     var manualParamChanges = null;
     var initStateName = null;
-    var version = GLOBALS.explorationVersion;
+    var version = UrlService.getExplorationVersionFromUrl();
+    if (version) {
+      ReadOnlyExplorationBackendApiService
+        .loadExploration(_explorationId, version)
+        .then(function(exploration) {
+          version = exploration.version;
+        });
+    } else {
+      ReadOnlyExplorationBackendApiService
+        .loadExploration(_explorationId)
+        .then(function(exploration) {
+          version = exploration.version;
+        });
+    }
```

Comment:
You don't need to have the `if` here, you can just call `.loadExploration(_explorationId, version)` (even if the version is null).